extensions/metrics/src/__tests__/acceptance/mock-pushgateway.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/extension-metrics
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import * as express from 'express';
+import {Server} from 'http';
+import pEvent from 'p-event';
+
+/**
+ * A mockup server for https://github.com/prometheus/pushgateway
+ */
+export class PushGateway {
+  private server?: Server;
+  private metrics: string[] = [];
+
+  async start() {
+    const app = express();
+    // A hack to force content-type to be `text/plain` as prom-client does not
+    // set `Content-Type` header
+    app.use((req, res, next) => {
+      req.headers['content-type'] = 'text/plain';
+      next();
+    });
+    app.use(express.text({type: '*/*'}));
+    app.get('/metrics', (req, res) => {
+      res.send(this.metrics.join('\n'));
+    });
+
+    app.post('/metrics/job/:jobName', (req, res) => {
+      this.metrics.push(`job="${req.params.jobName}"\n${req.body}`);
+      res.send('\n');
+    });
+    this.server = app.listen(9091);
```

Comment:
We should not use hard-coded port numbers, they can fail when tests are run in parallel or if the developer running the tests locally happen to have another process already listening on the same port.

You can use `app.listen(0)` or perhaps our `HttpServer` class.

```ts
import {HttpServer} from '@loopback/http-server';

this.server = new HttpServer(app, {/* no options are needed */});
await this.server.start();
```

Reply:
Unless the port `9091` is required by Prometheus and cannot be customized?